### DISCUSSION TITLE
How to restart a etcd pod in milvus cluster when it crash

### DISCUSSION THREAD

**Discussion by StormstoutY** [milvus-io/milvus#27702] [Q&A and General discussion]
I deploy a milvus cluster on a k8s cluster using helm chart v4.0.31 with only changing the milvus version to 2.3.1. After a while, I found that an etcd node crashed, and it continued to restart fail for multiple times. After that, I tried to expand an etcd node, but I still encountered the same error message when starting the container. And I found that the k8s nodes where two pod is deployed always has alarm messages like "failed to garbage collect required amount of images. Wanted to free 1782979788 bytes, but freed 0 bytes".

Is there any solution in this case? I found this doc https://milvus.io/docs/troubleshooting.md, will this solution cause the metadata stored on the etcd node by milvus to be lost?

Here is the error log of the crash etcd pod:
{"level":"info","ts":"2023-10-13T08:46:00.194Z","caller":"embed/etcd.go:306","msg":"starting an etcd server","etcd-version":"3.5.5","git-sha":"19002cfc6","go-version":"go1.16.15","go-os":"linux","go-arch":"amd64","max-cpu-set":16,"max-cpu-available":16,"member-initialized":false,......,"initial-cluster":"milvus-server-etcd-0=http://milvus-server-etcd-0.milvus-server-etcd-headless.default.svc.cluster.local:2380,milvus-server-etcd-1=http://milvus-server-etcd-1.milvus-server-etcd-headless.default.svc.cluster.local:2380,milvus-server-etcd-2=http://milvus-server-etcd-2.milvus-server-etcd-headless.default.svc.cluster.local:2380","initial-cluster-state":"new","initial-cluster-token":"etcd-cluster-k8s","quota-backend-bytes":4294967296,"max-request-bytes":1572864,"max-concurrent-streams":4294967295,"pre-vote":true,"initial-corrupt-check":false,"corrupt-check-time-interval":"0s","compact-check-time-enabled":false,"compact-check-time-interval":"1m0s","auto-compaction-mode":"revision","auto-compaction-retention":"1µs","auto-compaction-interval":"1µs","discovery-url":"","discovery-proxy":"","downgrade-check-interval":"5s"}

{"level":"info","ts":"2023-10-13T08:46:00.194Z","caller":"etcdserver/backend.go:81","msg":"opened backend db","path":"/bitnami/etcd/data/member/snap/db","took":"124.765µs"}

{"level":"info","ts":"2023-10-13T08:46:00.194Z","caller":"netutil/netutil.go:112","msg":"resolved URL Host","url":"http://milvus-server-etcd-1.milvus-server-etcd-headless.default.svc.cluster.local:2380/","host":"milvus-server-etcd-1.milvus-server-etcd-headless.default.svc.cluster.local:2380","resolved-addr":"1.2.3.4:2380"}

{"level":"info","ts":"2023-10-13T08:46:00.196Z","caller":"embed/etcd.go:371","msg":"closing etcd server","name":"milvus-server-etcd-1","data-dir":"/bitnami/etcd/data","advertise-peer-urls":["http://milvus-server-etcd-1.milvus-server-etcd-headless.default.svc.cluster.local:2380/"],"advertise-client-urls":["http://milvus-server-etcd-1.milvus-server-etcd-headless.default.svc.cluster.local:2379/"]}

{"level":"fatal","ts":"2023-10-13T08:46:00.196Z","caller":"etcdmain/etcd.go:204","msg":"discovery failed","error":"member 96c3b96d014ea9c has already been bootstrapped","stacktrace":"go.etcd.io/etcd/server/v3/etcdmain.startEtcdOrProxyV2\n\t/tmp/etcd-release-3.5.5/etcd/release/etcd/server/etcdmain/etcd.go:204\ngo.etcd.io/etcd/server/v3/etcdmain.Main\n\t/tmp/etcd-release-3.5.5/etcd/release/etcd/server/etcdmain/main.go:40\nmain.main\n\t/tmp/etcd-release-3.5.5/etcd/release/etcd/server/main.go:32\nruntime.main\n\t/usr/local/google/home/siarkowicz/.gvm/gos/go1.16.15/src/runtime/proc.go:225"}

**Comment by yhmo** [comment]
Search "failed to garbage collect required amount of images" on google, it is a k8s error message. A possible reason is it runs out of disk.
https://github.com/kubernetes/kubernetes/issues/71869

**Reply by caozhuozi** [reply]
@yhmo ,
I thinks the root cause may be related to this error:
{"level":"fatal","ts":"2023-10-13T08:46:00.196Z","caller":"etcdmain/etcd.go:204","msg":"discovery failed","error":"member 96c3b96d014ea9c has already been bootstrapped"
?

**Reply by filip-halt** [reply]
https://github.com/etcd-io/etcd/issues/2780#issuecomment-98039339

This possibly might be it, the data that etcd was using might be corrupted.

**Reply by StormstoutY** [reply]
Yes I think that's a problem of etcd cluster, but the etcd problem is more difficult to solve. It is even difficult for me to locate whether the problem occurs due to the k8s cluster itself, the etcd cluster, or the way milvus uses etcd. Btw I think the solution provided by https://milvus.io/docs/troubleshooting.md is a little bit risky, at least on my cluster.
